File: src/models/Staff.js
/**
 * Staff Model - OOP approach
 */
export class Staff {
  constructor(
    staffId,
    name,
    email,
    phone,
    address,
    joiningDate,
    qualification
  ) {
    this.staffId = staffId;
    this.name = name;
    this.email = email;
    this.phone = phone;
    this.address = address;
    this.joiningDate = joiningDate;
    this.qualification = qualification;
  }

  getStaffId() {
    return this.staffId;
  }

  setStaffId(staffId) {
    this.staffId = staffId;
  }

  getName() {
    return this.name;
  }

  setName(name) {
    this.name = name;
  }

  getEmail() {
    return this.email;
  }

  setEmail(email) {
    this.email = email;
  }

  getPhone() {
    return this.phone;
  }

  setPhone(phone) {
    this.phone = phone;
  }

  getAddress() {
    return this.address;
  }

  setAddress(address) {
    this.address = address;
  }

  getQualification() {
    return this.qualification;
  }

  setQualification(qualification) {
    this.qualification = qualification;
  }

  getJoiningDate() {
    return this.joiningDate;
  }

  setJoiningDate(joiningDate) {
    this.joiningDate = joiningDate;
  }

  toJSON() {
    return {
      staffId: this.staffId,
      name: this.name,
      email: this.email,
      phone: this.phone,
      address: this.address,
      joiningDate: this.joiningDate,
      qualification: this.qualification,
    };
  }
}

File: src/main/java/Class/Staff.java
package Class;

/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author nguye
 */
public class Staff {
    private static String staffId;
    private static String name;
    private static String email;
    private static String phone;
    private static String address;
    private static String qualification;
    private static String joiningDate;

    public String getJoiningDate() {
        return joiningDate;
    }

    public void setJoiningDate(String joiningDate) {
        Staff.joiningDate = joiningDate;
    }
    
    public Staff(){
        
    }

    public Staff(String staffId, String name, String email, String phone, String address, String joiningDate, String qualification) {
        this.staffId = staffId;
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.address = address;
        this.joiningDate = joiningDate;
        this.qualification = qualification;
    }

    public String getStaffId() {
        return staffId;
    }

    public void setStaffId(String staffId) {
        this.staffId = staffId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String adress) {
        this.address = adress;
    }

    public String getQualification() {
        return qualification;
    }

    public void setQualification(String qualification) {
        this.qualification = qualification;
    }

    
}
